# Kaggle TPS 202109

## Optuna LightGBM Starter

### Check GPU

In [ ]:
# Google Colab
gpu_info = !nvidia-smi
gpu_info = '\n'.join(gpu_info)
if gpu_info.find('failed') >= 0:
  print('Not connected to a GPU')
else:
  print(gpu_info)

Sat Sep 18 04:24:55 2021       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 470.63.01    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   40C    P0    26W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-------

### Check Memory

In [ ]:
from psutil import virtual_memory
ram_gb = virtual_memory().total / 1e9
print('Your runtime has {:.1f} gigabytes of available RAM\n'.format(ram_gb))

if ram_gb < 20:
  print('Not using a high-RAM runtime')
else:
  print('You are using a high-RAM runtime!')

Your runtime has 54.8 gigabytes of available RAM

You are using a high-RAM runtime!


### PIP Installs

Uninstall the colab default lightgbm, since it is not set up for GPU

In [ ]:
!pip uninstall lightgbm


Found existing installation: lightgbm 2.2.3
Uninstalling lightgbm-2.2.3:
  Would remove:
    /usr/local/lib/python3.7/dist-packages/lightgbm-2.2.3.dist-info/*
    /usr/local/lib/python3.7/dist-packages/lightgbm/*
Proceed (y/n)? y
  Successfully uninstalled lightgbm-2.2.3


For now, git clone the lightgbm from the repo.  (There is possibly another GPU friendly install option, but it hasn't been verified to work)

Note that this will take up space, so this is being done before mounting Google Drive



In [ ]:
! git clone --recursive https://github.com/Microsoft/LightGBM
! cd LightGBM && rm -rf build && mkdir build && cd build && cmake -DUSE_GPU=1 ../../LightGBM && make -j4 && cd ../python-package && python3 setup.py install --precompile --gpu;    
# unfortunately I forgot where I got the above, so cannot give them adequate credit

Cloning into 'LightGBM'...
remote: Enumerating objects: 23494, done.
remote: Counting objects: 100% (1171/1171), done.
remote: Compressing objects: 100% (631/631), done.
remote: Total 23494 (delta 731), reused 847 (delta 519), pack-reused 22323
Receiving objects: 100% (23494/23494), 18.12 MiB | 23.97 MiB/s, done.
Resolving deltas: 100% (17124/17124), done.
Submodule 'include/boost/compute' (https://github.com/boostorg/compute) registered for path 'external_libs/compute'
Submodule 'eigen' (https://gitlab.com/libeigen/eigen.git) registered for path 'external_libs/eigen'
Submodule 'external_libs/fast_double_parser' (https://github.com/lemire/fast_double_parser.git) registered for path 'external_libs/fast_double_parser'
Submodule 'external_libs/fmt' (https://github.com/fmtlib/fmt.git) registered for path 'external_libs/fmt'
Cloning into '/content/LightGBM/external_libs/compute'...
remote: Enumerating objects: 21731, done.        
remote: Counting objects: 100% (3/3), done.        
remote: 

Other untried option

```!pip install lightgbm --install-option=--gpu```


In [ ]:
# Next two lines are for Google Colab GPU and working with AutoGluon
!pip uninstall -y mkl
!pip install --upgrade mxnet-cu100

!pip install optuna

Found existing installation: mkl 2019.0
Uninstalling mkl-2019.0:
  Successfully uninstalled mkl-2019.0
     |████████████████████████████████| 352.6 MB 5.6 kB/s 
  Attempting uninstall: graphviz
    Found existing installation: graphviz 0.10.1
    Uninstalling graphviz-0.10.1:
      Successfully uninstalled graphviz-0.10.1
     |████████████████████████████████| 302 kB 12.8 MB/s 
     |████████████████████████████████| 80 kB 11.1 MB/s 
     |████████████████████████████████| 208 kB 79.6 MB/s 
     |████████████████████████████████| 75 kB 4.9 MB/s 
     |████████████████████████████████| 144 kB 93.9 MB/s 
     |████████████████████████████████| 111 kB 98.5 MB/s 
     |████████████████████████████████| 49 kB 7.0 MB/s 
  Created wheel for pyperclip: filename=pyperclip-1.8.2-py3-none-any.whl size=11136 sha256=5776c43d764c1215150f047aec941d6e5de4d67fb32fee42c2c6a82bff328b18
  Stored in directory: /root/.cache/pip/wheels/9f/18/84/8f69f8b08169c7bae2dde6bd7daf0c19fca8c8e500ee620a28
Successfull

## Imports

In [ ]:
import random
import os
from google.colab import output
import optuna


from lightgbm import LGBMClassifier
import lightgbm as lgb


import pandas as pd

from sklearn.ensemble import StackingRegressor
from sklearn.model_selection import train_test_split, cross_val_score, KFold
from sklearn.linear_model import LogisticRegression, LogisticRegressionCV
import pickle
import numpy as np
from sklearn.experimental import enable_iterative_imputer
from sklearn.impute import IterativeImputer
from sklearn.metrics import roc_auc_score

#importing all the plot functions
from optuna.visualization import plot_edf
from optuna.visualization import plot_optimization_history
from optuna.visualization import plot_parallel_coordinate
from optuna.visualization import plot_param_importances
from optuna.visualization import plot_slice

import logging

## Google Drive Mount

* This is used to save models and output
* Note that for this Kaggle competition, the train.csv and test.csv are quite large and take a while to upload and store.

In [ ]:
working_dir = 'MyDrive/kaggle/202109'
drive_dir = '/content/drive/'
model_dir = 'models'
output_dir = 'output'
from google.colab import drive
drive.mount(drive_dir, force_remount=True)  # Do this once only

mounted_drive = f'{drive_dir}/{working_dir}' #''
os.chdir(mounted_drive)


Mounted at /content/drive/


In [ ]:
os.environ['KAGGLE_CONFIG_DIR'] = "/content/drive/MyDrive/.kaggle"

If you want to directly download the kaggle files to your Google Colab environment

From https://www.analyticsvidhya.com/blog/2021/05/10-colab-tips-and-hacks-for-efficient-use-of-it/

https://github.com/Kaggle/kaggle-api

You can pull the datasets directly from Kaggle
1.  Go to your account page on kaggle.com and scroll to API to get your token
2.  Download the token 
3.  Copy it to Google Drive - most people put it in the .kaggle directory
4.  Set enviornment variable KAGGLE_CONFIG_DIR
5.  ```
! pip install kaggle
kaggle competitions download -c tabular-playground-series-sep-2021```
6.  You may need to add logic to move the files where you want them

In [ ]:
#!pip install kaggle
#!kaggle competitions download -c tabular-playground-series-sep-2021

In [ ]:
pd.set_option("display.max_rows", 999)
pd.set_option("display.max_columns", 999)

# Read csv files
df = pd.read_csv('./train.csv')
test_df=pd.read_csv('./test.csv')


## Logging

In [ ]:
# Right now this is not working as expected, so this is pending
#logging.basicConfig(filename='./artifacts/Kaggle_202109_lgbm.log',level=logging.INFO)

# Massaging Data

These are taken from Kaggle notebooks (see commented code)


1.   Memory optimization 
     * scans each column and based on the numeric range, sets the type to the one that requires the least memory
     * improves performance because the datasets are very large
2.   Imputation
     * uses sklearn's iterative imputer which approximates the missing values based on "nearby" values.
     * improves scores
     * note that it takes a while, so for this notebook, the massaged data is being saved locally
3.   additional columns are added to indicate the number of rows missing data and the missing feature columns

In [ ]:
RUN_MASSAGE = False

In [ ]:
# https://www.kaggle.com/lucamassaron/autogluon-for-tabular-playground-sep-2021
# Derived from the original script https://www.kaggle.com/gemartin/load-data-reduce-memory-usage 
# by Guillaume Martin

# This checks the range of values in each column and resets the type 

def reduce_mem_usage(df, verbose=True):
    numerics = ['int16', 'int32', 'int64', 'float16', 'float32', 'float64']
    start_mem = df.memory_usage().sum() / 1024**2    
    for col in df.columns:
        col_type = df[col].dtypes
        if col_type in numerics:
            c_min = df[col].min()
            c_max = df[col].max()
            if str(col_type)[:3] == 'int':
                if c_min > np.iinfo(np.int8).min and c_max < np.iinfo(np.int8).max:
                    df[col] = df[col].astype(np.int8)
                elif c_min > np.iinfo(np.int16).min and c_max < np.iinfo(np.int16).max:
                    df[col] = df[col].astype(np.int16)
                elif c_min > np.iinfo(np.int32).min and c_max < np.iinfo(np.int32).max:
                    df[col] = df[col].astype(np.int32)
                elif c_min > np.iinfo(np.int64).min and c_max < np.iinfo(np.int64).max:
                    df[col] = df[col].astype(np.int64)  
            else:
                if c_min > np.finfo(np.float32).min and c_max < np.finfo(np.float32).max:
                    df[col] = df[col].astype(np.float32)
                else:
                    df[col] = df[col].astype(np.float64)    
    end_mem = df.memory_usage().sum() / 1024**2
    if verbose: print('Mem. usage decreased to {:5.2f} Mb ({:.1f}% reduction)'.format(end_mem, 100 * (start_mem - end_mem) / start_mem))
    return df

In [ ]:
df = reduce_mem_usage(df)
test_df = reduce_mem_usage(test_df)

Mem. usage decreased to 435.76 Mb (50.3% reduction)
Mem. usage decreased to 224.01 Mb (50.0% reduction)


In [ ]:
if RUN_MASSAGE:
  X = df.drop(columns = ['id', 'claim'])
  test_X = test_df.drop(columns = ['id'])

In [ ]:
#https://www.kaggle.com/lucamassaron/autogluon-for-tabular-playground-sep-2021
# Reference: https://www.kaggle.com/hsuchialun/tps-lightgbm-kfold
if RUN_MASSAGE:
    feats = list(X.columns[:-1])

    X['n_row_missing'] = X[feats].isna().sum(axis=1)
    test_X['n_row_missing'] = test_X[feats].isna().sum(axis=1)

    X['row_std'] = X[feats].std(axis=1)
    test_X['row_std'] = test_X[feats].std(axis=1)

    feats += ['n_row_missing', 'row_std']


In [ ]:
# MULTIVARIATE ITERATIVE IMPUTATION

if RUN_MASSAGE:
    
    imp = IterativeImputer(max_iter=10, random_state=0)
    imp.fit(X[feats].sample(n=10_000))

    X[feats] = imp.transform(X[feats])
    test_X[feats] = imp.transform(test_X[feats])

In [ ]:
if RUN_MASSAGE:
    pd.DataFrame(X).to_csv('./reduced_mem_imputed.csv', index=False)
    pd.DataFrame(test_X).to_csv('./reduced_mem_imputesd_test.csv', index=False)

In [ ]:
X = pd.read_csv('./reduced_mem_imputed.csv')
test_X = pd.read_csv('./reduced_mem_imputesd_test.csv') # this is the Kaggle test dataset

In [ ]:
y = df[['claim']]
N_SPLITS = 5 # if you're using cross_score_validation
EARLY_STOPPING_ROUNDS = 400
SEED = 42

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=42)

Optuna 
* study - this is the tuning done over several iterations of different hyperparameters
* trial - this is the individual evaluation/trial of selected hyperparameters
* objective function - defines the logic that Optuna will use for a single trial
* callback - allows actions after the objective function has returned the ranking metric to Optuna.  

At this point Optuna does not save the best model for use after the study.  Instead it will return the best parameters of the study.  However the study can take a long time, so a callback is used to save the best model (as of yet) to a file.

**Note that there should be a sample file called best_{model_type} which will be replaced by the best model of the study.  Right now the job will err out, if the model isn't there. ** (And yes, more code can be written to automatically create that file or check it's existence, but it hasn't been done yet)

Due to the large training and test datasets, the objective function has been modified to work with train_test_split

In [ ]:
def lgbm_objective(trial):
   
    # hyperparameter setting, trial.suggest_uniform will suggest uniform hyperparameter
    
    
    kfolds = KFold(n_splits=N_SPLITS, shuffle=True)
    
    params = {
        "metric": "AUC",
        "verbosity": 0,
        "device": "gpu",
        "boosting_type": trial.suggest_categorical('boosting_type', ['gbdt']), # , 'dart'
        "subsample": trial.suggest_uniform('subsample', 0.5, 0.6),
        "learning_rate": trial.suggest_float('eta', 0.005, 0.015),
        "num_iterations": trial.suggest_int('num_iterations', 1000, 40000),
        "subsample_freq": 1,
        "lambda_l1": trial.suggest_float("lambda_l1", 1e-7, 10.0, log=True),
        "lambda_l2": trial.suggest_float("lambda_l2", 1e-7, 10.0, log=True),
        "num_leaves": trial.suggest_int("num_leaves", 2, 256),
        "feature_fraction": trial.suggest_float("feature_fraction", 0.4, 1.0),        
        "min_child_samples": trial.suggest_int("min_child_samples", 5, 100),
    }
    
    print(f'STARTING Params: {params}')
    lgbm_model = LGBMClassifier(**params,random_state=42)
    
    d_train = lgb.Dataset(X, label=y)

    #https://www.kaggle.com/aharless/trying-to-do-k-fold-cv-on-lightgbm
    cv_results = lgb.cv(params, d_train, early_stopping_rounds=EARLY_STOPPING_ROUNDS, nfold=N_SPLITS, metrics = 'auc', seed=42)
    
    #print(cv_results)
    #results['auc-mean'][-1]
    auc = cv_results['auc-mean']
    print('Trial LGBM AUC',np.amax(auc), np.amin(auc), np.sqrt(abs(np.mean(auc))))
    filename = f'./models/lgbm_{trial.number}'
    
    pickle.dump(lgbm_model, open(filename, 'wb'))
    
    return auc[-1] 


def lgbm_callback(study, trial):
  # from os import remove
    filename = f'./models/lgbm_{trial.number}'
    
    best_model_name = './models/best_lgbm'
    if study.best_trial.number == trial.number:
        os.remove(best_model_name)
        os.rename(filename, best_model_name)
    else:
        os.remove(filename)
    print(f'filename {filename} best_model_name{best_model_name} ; best_trial_number {study.best_trial.number}')

In [ ]:
lgbm_study = optuna.create_study(direction='maximize')
lgbm_study.optimize(lgbm_objective, n_trials=100, callbacks=[lgbm_callback])

[I 2021-09-17 06:09:29,251] A new study created in memory with name: no-name-05115bdc-059b-4c5b-b09f-938889ea253a
/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.



STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5727063392015879, 'learning_rate': 0.010278674171839874, 'num_iterations': 1569, 'subsample_freq': 1, 'lambda_l1': 0.09211554066717548, 'lambda_l2': 0.12079930458940936, 'num_leaves': 162, 'feature_fraction': 0.7936490021224217, 'min_child_samples': 66}


[I 2021-09-17 06:18:36,096] Trial 0 finished with value: 0.8146050196152306 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5727063392015879, 'eta': 0.010278674171839874, 'num_iterations': 1569, 'lambda_l1': 0.09211554066717548, 'lambda_l2': 0.12079930458940936, 'num_leaves': 162, 'feature_fraction': 0.7936490021224217, 'min_child_samples': 66}. Best is trial 0 with value: 0.8146050196152306.


Trial LGBM AUC 0.8146050196152306 0.5408594042261294 0.9013952571080929
filename ./models/lgbm_0 best_model_name./models/best_lgbm ; best_trial_number 0
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5723089372660386, 'learning_rate': 0.007122586554732636, 'num_iterations': 12974, 'subsample_freq': 1, 'lambda_l1': 0.5901477129019864, 'lambda_l2': 1.8336514748901417e-05, 'num_leaves': 228, 'feature_fraction': 0.5988502385158633, 'min_child_samples': 25}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 06:47:44,328] Trial 1 finished with value: 0.8149008263401365 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5723089372660386, 'eta': 0.007122586554732636, 'num_iterations': 12974, 'lambda_l1': 0.5901477129019864, 'lambda_l2': 1.8336514748901417e-05, 'num_leaves': 228, 'feature_fraction': 0.5988502385158633, 'min_child_samples': 25}. Best is trial 1 with value: 0.8149008263401365.


Trial LGBM AUC 0.8149008263401365 0.5417103347997344 0.9020221373560633
filename ./models/lgbm_1 best_model_name./models/best_lgbm ; best_trial_number 1
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5126628639792093, 'learning_rate': 0.0061624207428288644, 'num_iterations': 22354, 'subsample_freq': 1, 'lambda_l1': 1.671310439721814e-05, 'lambda_l2': 2.901942685738539e-06, 'num_leaves': 48, 'feature_fraction': 0.6174261167967579, 'min_child_samples': 98}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 07:14:50,210] Trial 2 finished with value: 0.8157054565554273 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5126628639792093, 'eta': 0.0061624207428288644, 'num_iterations': 22354, 'lambda_l1': 1.671310439721814e-05, 'lambda_l2': 2.901942685738539e-06, 'num_leaves': 48, 'feature_fraction': 0.6174261167967579, 'min_child_samples': 98}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8157054565554273 0.542873315565455 0.9025394387812226
filename ./models/lgbm_2 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5636025869136195, 'learning_rate': 0.007446201811739935, 'num_iterations': 18029, 'subsample_freq': 1, 'lambda_l1': 0.10208897070895423, 'lambda_l2': 5.0475795956204396e-05, 'num_leaves': 239, 'feature_fraction': 0.4121270422539963, 'min_child_samples': 56}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 07:46:48,838] Trial 3 finished with value: 0.8152006805647668 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5636025869136195, 'eta': 0.007446201811739935, 'num_iterations': 18029, 'lambda_l1': 0.10208897070895423, 'lambda_l2': 5.0475795956204396e-05, 'num_leaves': 239, 'feature_fraction': 0.4121270422539963, 'min_child_samples': 56}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8152006805647668 0.5409810823206036 0.9020990315892111
filename ./models/lgbm_3 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5007860424564303, 'learning_rate': 0.011939819737369326, 'num_iterations': 32718, 'subsample_freq': 1, 'lambda_l1': 0.0017706437978294663, 'lambda_l2': 0.012768654749150034, 'num_leaves': 240, 'feature_fraction': 0.9679895939300337, 'min_child_samples': 12}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 08:03:43,492] Trial 4 finished with value: 0.8139649390431956 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5007860424564303, 'eta': 0.011939819737369326, 'num_iterations': 32718, 'lambda_l1': 0.0017706437978294663, 'lambda_l2': 0.012768654749150034, 'num_leaves': 240, 'feature_fraction': 0.9679895939300337, 'min_child_samples': 12}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8139649390431956 0.7992471449703173 0.9015322132181173
filename ./models/lgbm_4 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.596336469617212, 'learning_rate': 0.013885929020462089, 'num_iterations': 6458, 'subsample_freq': 1, 'lambda_l1': 7.752932300497012e-05, 'lambda_l2': 0.04070683497888291, 'num_leaves': 44, 'feature_fraction': 0.8167298690593833, 'min_child_samples': 21}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 08:16:06,461] Trial 5 finished with value: 0.8150371753786638 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.596336469617212, 'eta': 0.013885929020462089, 'num_iterations': 6458, 'lambda_l1': 7.752932300497012e-05, 'lambda_l2': 0.04070683497888291, 'num_leaves': 44, 'feature_fraction': 0.8167298690593833, 'min_child_samples': 21}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8150371753786638 0.5444846591223156 0.9022568953903441
filename ./models/lgbm_5 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5574441028378082, 'learning_rate': 0.008646103619500928, 'num_iterations': 21915, 'subsample_freq': 1, 'lambda_l1': 0.08307995752985149, 'lambda_l2': 9.363303949845661e-07, 'num_leaves': 2, 'feature_fraction': 0.9151171172110089, 'min_child_samples': 97}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 08:43:49,043] Trial 6 finished with value: 0.8136138909271631 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5574441028378082, 'eta': 0.008646103619500928, 'num_iterations': 21915, 'lambda_l1': 0.08307995752985149, 'lambda_l2': 9.363303949845661e-07, 'num_leaves': 2, 'feature_fraction': 0.9151171172110089, 'min_child_samples': 97}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8136138909271631 0.507735461595265 0.9006968200604644
filename ./models/lgbm_6 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5679608194386208, 'learning_rate': 0.009257613888755923, 'num_iterations': 15956, 'subsample_freq': 1, 'lambda_l1': 9.402307740854103e-05, 'lambda_l2': 0.5624867081529631, 'num_leaves': 117, 'feature_fraction': 0.46640791076951044, 'min_child_samples': 84}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 09:01:04,823] Trial 7 finished with value: 0.8153197888872864 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5679608194386208, 'eta': 0.009257613888755923, 'num_iterations': 15956, 'lambda_l1': 9.402307740854103e-05, 'lambda_l2': 0.5624867081529631, 'num_leaves': 117, 'feature_fraction': 0.46640791076951044, 'min_child_samples': 84}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8153197888872864 0.5414124940168691 0.9020325513668316
filename ./models/lgbm_7 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5974731049898263, 'learning_rate': 0.014167464784756376, 'num_iterations': 17694, 'subsample_freq': 1, 'lambda_l1': 3.9118381575739135e-07, 'lambda_l2': 0.022750853084495065, 'num_leaves': 24, 'feature_fraction': 0.5938132098123698, 'min_child_samples': 48}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 09:17:00,041] Trial 8 finished with value: 0.8155502074006629 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5974731049898263, 'eta': 0.014167464784756376, 'num_iterations': 17694, 'lambda_l1': 3.9118381575739135e-07, 'lambda_l2': 0.022750853084495065, 'num_leaves': 24, 'feature_fraction': 0.5938132098123698, 'min_child_samples': 48}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8155502074006629 0.5406170344126303 0.9025723561572209
filename ./models/lgbm_8 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.53662188838731, 'learning_rate': 0.009060770761417349, 'num_iterations': 13158, 'subsample_freq': 1, 'lambda_l1': 0.00094468386535772, 'lambda_l2': 0.00043148373240762937, 'num_leaves': 99, 'feature_fraction': 0.4097233939419307, 'min_child_samples': 13}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 09:34:00,059] Trial 9 finished with value: 0.8151673774578466 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.53662188838731, 'eta': 0.009060770761417349, 'num_iterations': 13158, 'lambda_l1': 0.00094468386535772, 'lambda_l2': 0.00043148373240762937, 'num_leaves': 99, 'feature_fraction': 0.4097233939419307, 'min_child_samples': 13}. Best is trial 2 with value: 0.8157054565554273.


Trial LGBM AUC 0.8151673774578466 0.5426591887221252 0.9019971822694327
filename ./models/lgbm_9 best_model_name./models/best_lgbm ; best_trial_number 2
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5017593506088529, 'learning_rate': 0.005403931043209345, 'num_iterations': 29381, 'subsample_freq': 1, 'lambda_l1': 1.7336997491412977e-07, 'lambda_l2': 1.1829633575532706e-07, 'num_leaves': 69, 'feature_fraction': 0.6573049536503066, 'min_child_samples': 99}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 10:15:07,169] Trial 10 finished with value: 0.8157128828715831 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5017593506088529, 'eta': 0.005403931043209345, 'num_iterations': 29381, 'lambda_l1': 1.7336997491412977e-07, 'lambda_l2': 1.1829633575532706e-07, 'num_leaves': 69, 'feature_fraction': 0.6573049536503066, 'min_child_samples': 99}. Best is trial 10 with value: 0.8157128828715831.


Trial LGBM AUC 0.8157128828715831 0.5426471813237931 0.9026632501551617
filename ./models/lgbm_10 best_model_name./models/best_lgbm ; best_trial_number 10
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5019406654813531, 'learning_rate': 0.005172895979064616, 'num_iterations': 29362, 'subsample_freq': 1, 'lambda_l1': 1.5040990005655627e-07, 'lambda_l2': 1.2110130371977854e-07, 'num_leaves': 69, 'feature_fraction': 0.6160469875710929, 'min_child_samples': 96}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 10:50:43,540] Trial 11 finished with value: 0.8157812451608006 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5019406654813531, 'eta': 0.005172895979064616, 'num_iterations': 29362, 'lambda_l1': 1.5040990005655627e-07, 'lambda_l2': 1.2110130371977854e-07, 'num_leaves': 69, 'feature_fraction': 0.6160469875710929, 'min_child_samples': 96}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8157812451608006 0.5422841630188685 0.9025939998662025
filename ./models/lgbm_11 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5259194267722721, 'learning_rate': 0.00527276508862559, 'num_iterations': 33609, 'subsample_freq': 1, 'lambda_l1': 1.1435222044985643e-07, 'lambda_l2': 1.0122585280364721e-07, 'num_leaves': 79, 'feature_fraction': 0.6987533311023189, 'min_child_samples': 78}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 11:21:17,207] Trial 12 finished with value: 0.8155551295048993 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5259194267722721, 'eta': 0.00527276508862559, 'num_iterations': 33609, 'lambda_l1': 1.1435222044985643e-07, 'lambda_l2': 1.0122585280364721e-07, 'num_leaves': 79, 'feature_fraction': 0.6987533311023189, 'min_child_samples': 78}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8155551295048993 0.542531854372488 0.9024435064906074
filename ./models/lgbm_12 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5024037022003133, 'learning_rate': 0.005133960323636035, 'num_iterations': 27946, 'subsample_freq': 1, 'lambda_l1': 2.2942740302876687e-06, 'lambda_l2': 1.0789339643868345e-07, 'num_leaves': 160, 'feature_fraction': 0.7118508416702233, 'min_child_samples': 80}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 12:03:46,516] Trial 13 finished with value: 0.8154185353062167 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5024037022003133, 'eta': 0.005133960323636035, 'num_iterations': 27946, 'lambda_l1': 2.2942740302876687e-06, 'lambda_l2': 1.0789339643868345e-07, 'num_leaves': 160, 'feature_fraction': 0.7118508416702233, 'min_child_samples': 80}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8154185353062167 0.540987963515118 0.9024338558988458
filename ./models/lgbm_13 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.524210217320848, 'learning_rate': 0.006851098465148982, 'num_iterations': 39554, 'subsample_freq': 1, 'lambda_l1': 1.3200559470570801e-06, 'lambda_l2': 1.4405933920487736e-06, 'num_leaves': 78, 'feature_fraction': 0.5405579875345325, 'min_child_samples': 98}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 12:31:05,308] Trial 14 finished with value: 0.8156249758574171 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.524210217320848, 'eta': 0.006851098465148982, 'num_iterations': 39554, 'lambda_l1': 1.3200559470570801e-06, 'lambda_l2': 1.4405933920487736e-06, 'num_leaves': 78, 'feature_fraction': 0.5405579875345325, 'min_child_samples': 98}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8156249758574171 0.5430916867443718 0.902463268509632
filename ./models/lgbm_14 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5402444390845318, 'learning_rate': 0.011092160577207397, 'num_iterations': 26730, 'subsample_freq': 1, 'lambda_l1': 7.735996059669064e-06, 'lambda_l2': 0.0006878621166496951, 'num_leaves': 134, 'feature_fraction': 0.7001145249591282, 'min_child_samples': 46}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 12:47:51,281] Trial 15 finished with value: 0.8147710856980008 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5402444390845318, 'eta': 0.011092160577207397, 'num_iterations': 26730, 'lambda_l1': 7.735996059669064e-06, 'lambda_l2': 0.0006878621166496951, 'num_leaves': 134, 'feature_fraction': 0.7001145249591282, 'min_child_samples': 46}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8147710856980008 0.5417071441995484 0.9020591666253103
filename ./models/lgbm_15 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5171274348797861, 'learning_rate': 0.008004950128349248, 'num_iterations': 35800, 'subsample_freq': 1, 'lambda_l1': 1.0533987569539099e-07, 'lambda_l2': 2.3466610467039646e-05, 'num_leaves': 67, 'feature_fraction': 0.5332882298873162, 'min_child_samples': 67}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 13:14:33,184] Trial 16 finished with value: 0.8154891262308869 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5171274348797861, 'eta': 0.008004950128349248, 'num_iterations': 35800, 'lambda_l1': 1.0533987569539099e-07, 'lambda_l2': 2.3466610467039646e-05, 'num_leaves': 67, 'feature_fraction': 0.5332882298873162, 'min_child_samples': 67}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8154891262308869 0.5420146534003578 0.9024785429380229
filename ./models/lgbm_16 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5115508515462496, 'learning_rate': 0.005985986840097134, 'num_iterations': 27518, 'subsample_freq': 1, 'lambda_l1': 0.0022511125488097485, 'lambda_l2': 6.948364002321384e-07, 'num_leaves': 144, 'feature_fraction': 0.7917098220797911, 'min_child_samples': 89}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 13:49:03,545] Trial 17 finished with value: 0.8154202854446935 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5115508515462496, 'eta': 0.005985986840097134, 'num_iterations': 27518, 'lambda_l1': 0.0022511125488097485, 'lambda_l2': 6.948364002321384e-07, 'num_leaves': 144, 'feature_fraction': 0.7917098220797911, 'min_child_samples': 89}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8154202854446935 0.5425331586381169 0.9024076580440012
filename ./models/lgbm_17 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.53833874888673, 'learning_rate': 0.012604279892076194, 'num_iterations': 30267, 'subsample_freq': 1, 'lambda_l1': 1.3463543290939029e-06, 'lambda_l2': 3.612743718988148e-06, 'num_leaves': 195, 'feature_fraction': 0.6539172744436095, 'min_child_samples': 71}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 14:00:54,279] Trial 18 finished with value: 0.8145462725082278 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.53833874888673, 'eta': 0.012604279892076194, 'num_iterations': 30267, 'lambda_l1': 1.3463543290939029e-06, 'lambda_l2': 3.612743718988148e-06, 'num_leaves': 195, 'feature_fraction': 0.6539172744436095, 'min_child_samples': 71}. Best is trial 11 with value: 0.8157812451608006.


Trial LGBM AUC 0.8145462725082278 0.542139713427313 0.9015690757029317
filename ./models/lgbm_18 best_model_name./models/best_lgbm ; best_trial_number 11
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.526377224809406, 'learning_rate': 0.0062010318996095016, 'num_iterations': 39614, 'subsample_freq': 1, 'lambda_l1': 9.79895795913152, 'lambda_l2': 2.6293226443181807, 'num_leaves': 100, 'feature_fraction': 0.5228755040643674, 'min_child_samples': 40}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 14:32:52,914] Trial 19 finished with value: 0.8158655110229824 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.526377224809406, 'eta': 0.0062010318996095016, 'num_iterations': 39614, 'lambda_l1': 9.79895795913152, 'lambda_l2': 2.6293226443181807, 'num_leaves': 100, 'feature_fraction': 0.5228755040643674, 'min_child_samples': 40}. Best is trial 19 with value: 0.8158655110229824.


Trial LGBM AUC 0.8158655110229824 0.5422660941957943 0.902297571654655
filename ./models/lgbm_19 best_model_name./models/best_lgbm ; best_trial_number 19
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5286474663251459, 'learning_rate': 0.0063520792774278585, 'num_iterations': 39718, 'subsample_freq': 1, 'lambda_l1': 4.024804310483001, 'lambda_l2': 1.0822805234205795, 'num_leaves': 108, 'feature_fraction': 0.49815973796353064, 'min_child_samples': 36}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 15:07:12,974] Trial 20 finished with value: 0.8159092406767193 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5286474663251459, 'eta': 0.0063520792774278585, 'num_iterations': 39718, 'lambda_l1': 4.024804310483001, 'lambda_l2': 1.0822805234205795, 'num_leaves': 108, 'feature_fraction': 0.49815973796353064, 'min_child_samples': 36}. Best is trial 20 with value: 0.8159092406767193.


Trial LGBM AUC 0.8159092406767193 0.541571955242931 0.9024998119014747
filename ./models/lgbm_20 best_model_name./models/best_lgbm ; best_trial_number 20
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5278299681437211, 'learning_rate': 0.006487677079659437, 'num_iterations': 39520, 'subsample_freq': 1, 'lambda_l1': 9.014865740694933, 'lambda_l2': 4.245358888340319, 'num_leaves': 99, 'feature_fraction': 0.5047457136539589, 'min_child_samples': 39}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 15:41:55,248] Trial 21 finished with value: 0.8159660099828953 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5278299681437211, 'eta': 0.006487677079659437, 'num_iterations': 39520, 'lambda_l1': 9.014865740694933, 'lambda_l2': 4.245358888340319, 'num_leaves': 99, 'feature_fraction': 0.5047457136539589, 'min_child_samples': 39}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8159660099828953 0.5418308007020542 0.9024664692417133
filename ./models/lgbm_21 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5476151240163777, 'learning_rate': 0.006737401156176602, 'num_iterations': 39304, 'subsample_freq': 1, 'lambda_l1': 9.085590519618057, 'lambda_l2': 3.7827007575533216, 'num_leaves': 103, 'feature_fraction': 0.5002770482278937, 'min_child_samples': 34}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 16:14:23,204] Trial 22 finished with value: 0.8158916849005287 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5476151240163777, 'eta': 0.006737401156176602, 'num_iterations': 39304, 'lambda_l1': 9.085590519618057, 'lambda_l2': 3.7827007575533216, 'num_leaves': 103, 'feature_fraction': 0.5002770482278937, 'min_child_samples': 34}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8158916849005287 0.5413524224084356 0.9024078459035967
filename ./models/lgbm_22 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5489213569373916, 'learning_rate': 0.008008751277349029, 'num_iterations': 36280, 'subsample_freq': 1, 'lambda_l1': 9.033502306120315, 'lambda_l2': 8.834261990773554, 'num_leaves': 108, 'feature_fraction': 0.48467504526475336, 'min_child_samples': 36}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 16:44:00,396] Trial 23 finished with value: 0.815836401832903 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5489213569373916, 'eta': 0.008008751277349029, 'num_iterations': 36280, 'lambda_l1': 9.033502306120315, 'lambda_l2': 8.834261990773554, 'num_leaves': 108, 'feature_fraction': 0.48467504526475336, 'min_child_samples': 36}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.815836401832903 0.5418476631258802 0.9024070581934397
filename ./models/lgbm_23 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5458952203761629, 'learning_rate': 0.006682554523136759, 'num_iterations': 36637, 'subsample_freq': 1, 'lambda_l1': 1.5666594925545594, 'lambda_l2': 0.6403096406624899, 'num_leaves': 172, 'feature_fraction': 0.4850439676652866, 'min_child_samples': 32}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 17:20:50,927] Trial 24 finished with value: 0.8155692177632708 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5458952203761629, 'eta': 0.006682554523136759, 'num_iterations': 36637, 'lambda_l1': 1.5666594925545594, 'lambda_l2': 0.6403096406624899, 'num_leaves': 172, 'feature_fraction': 0.4850439676652866, 'min_child_samples': 32}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8155692177632708 0.5416924113517304 0.9023790854520122
filename ./models/lgbm_24 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5303674526915412, 'learning_rate': 0.00780154540223571, 'num_iterations': 38376, 'subsample_freq': 1, 'lambda_l1': 1.1082998631419592, 'lambda_l2': 9.85949255394303, 'num_leaves': 128, 'feature_fraction': 0.4587223596168046, 'min_child_samples': 57}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 17:43:10,960] Trial 25 finished with value: 0.8156485926046887 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5303674526915412, 'eta': 0.00780154540223571, 'num_iterations': 38376, 'lambda_l1': 1.1082998631419592, 'lambda_l2': 9.85949255394303, 'num_leaves': 128, 'feature_fraction': 0.4587223596168046, 'min_child_samples': 57}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8156485926046887 0.5412928315843264 0.9021179005410486
filename ./models/lgbm_25 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5545466973928422, 'learning_rate': 0.00982204963314236, 'num_iterations': 33736, 'subsample_freq': 1, 'lambda_l1': 2.992990005859538, 'lambda_l2': 0.6459315356526913, 'num_leaves': 93, 'feature_fraction': 0.5619844890273882, 'min_child_samples': 26}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 17:59:17,937] Trial 26 finished with value: 0.8155040270732249 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5545466973928422, 'eta': 0.00982204963314236, 'num_iterations': 33736, 'lambda_l1': 2.992990005859538, 'lambda_l2': 0.6459315356526913, 'num_leaves': 93, 'feature_fraction': 0.5619844890273882, 'min_child_samples': 26}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8155040270732249 0.5428896483306029 0.902032045382859
filename ./models/lgbm_26 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5345397986250907, 'learning_rate': 0.006261299680087225, 'num_iterations': 25145, 'subsample_freq': 1, 'lambda_l1': 0.37980979670350395, 'lambda_l2': 0.0033863472492283125, 'num_leaves': 191, 'feature_fraction': 0.4329496020649507, 'min_child_samples': 41}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 18:30:04,062] Trial 27 finished with value: 0.8153620432865765 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5345397986250907, 'eta': 0.006261299680087225, 'num_iterations': 25145, 'lambda_l1': 0.37980979670350395, 'lambda_l2': 0.0033863472492283125, 'num_leaves': 191, 'feature_fraction': 0.4329496020649507, 'min_child_samples': 41}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8153620432865765 0.5419741912239635 0.9020925306314148
filename ./models/lgbm_27 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5438042512574957, 'learning_rate': 0.008528908753138347, 'num_iterations': 31570, 'subsample_freq': 1, 'lambda_l1': 0.015315221103088441, 'lambda_l2': 0.14928919400138999, 'num_leaves': 123, 'feature_fraction': 0.5079540484159953, 'min_child_samples': 32}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 18:51:23,142] Trial 28 finished with value: 0.8152354194436203 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5438042512574957, 'eta': 0.008528908753138347, 'num_iterations': 31570, 'lambda_l1': 0.015315221103088441, 'lambda_l2': 0.14928919400138999, 'num_leaves': 123, 'feature_fraction': 0.5079540484159953, 'min_child_samples': 32}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8152354194436203 0.5422689528359176 0.9020982564339145
filename ./models/lgbm_28 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5199615493327984, 'learning_rate': 0.007230557119618307, 'num_iterations': 1948, 'subsample_freq': 1, 'lambda_l1': 0.1685291461724972, 'lambda_l2': 1.6960529374481275, 'num_leaves': 150, 'feature_fraction': 0.5654912102280674, 'min_child_samples': 6}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 19:01:53,565] Trial 29 finished with value: 0.8146042139555598 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5199615493327984, 'eta': 0.007230557119618307, 'num_iterations': 1948, 'lambda_l1': 0.1685291461724972, 'lambda_l2': 1.6960529374481275, 'num_leaves': 150, 'feature_fraction': 0.5654912102280674, 'min_child_samples': 6}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8146042139555598 0.54219086785774 0.9011014910957529
filename ./models/lgbm_29 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5561763197913913, 'learning_rate': 0.005938065856971338, 'num_iterations': 35649, 'subsample_freq': 1, 'lambda_l1': 0.009458669230133518, 'lambda_l2': 0.15401873637098, 'num_leaves': 53, 'feature_fraction': 0.7544203019616958, 'min_child_samples': 53}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 19:33:09,453] Trial 30 finished with value: 0.815625359448352 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5561763197913913, 'eta': 0.005938065856971338, 'num_iterations': 35649, 'lambda_l1': 0.009458669230133518, 'lambda_l2': 0.15401873637098, 'num_leaves': 53, 'feature_fraction': 0.7544203019616958, 'min_child_samples': 53}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.815625359448352 0.5417016960761527 0.9025494297861603
filename ./models/lgbm_30 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5831588943568513, 'learning_rate': 0.00663997823891355, 'num_iterations': 39786, 'subsample_freq': 1, 'lambda_l1': 9.17935937317507, 'lambda_l2': 2.651315984269685, 'num_leaves': 91, 'feature_fraction': 0.5173521241641071, 'min_child_samples': 41}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 20:05:21,535] Trial 31 finished with value: 0.8159224372416475 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5831588943568513, 'eta': 0.00663997823891355, 'num_iterations': 39786, 'lambda_l1': 9.17935937317507, 'lambda_l2': 2.651315984269685, 'num_leaves': 91, 'feature_fraction': 0.5173521241641071, 'min_child_samples': 41}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8159224372416475 0.5425131193995053 0.9024250385627037
filename ./models/lgbm_31 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5890952322822424, 'learning_rate': 0.006696600376128748, 'num_iterations': 39692, 'subsample_freq': 1, 'lambda_l1': 3.56953478069416, 'lambda_l2': 2.263719207938312, 'num_leaves': 93, 'feature_fraction': 0.501014307073317, 'min_child_samples': 25}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 20:42:07,155] Trial 32 finished with value: 0.8158985730677593 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5890952322822424, 'eta': 0.006696600376128748, 'num_iterations': 39692, 'lambda_l1': 3.56953478069416, 'lambda_l2': 2.263719207938312, 'num_leaves': 93, 'feature_fraction': 0.501014307073317, 'min_child_samples': 25}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8158985730677593 0.5414247609896364 0.9026042759496239
filename ./models/lgbm_32 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5872409883213553, 'learning_rate': 0.007405870921478667, 'num_iterations': 37569, 'subsample_freq': 1, 'lambda_l1': 2.8232682174433457, 'lambda_l2': 0.33103371803648324, 'num_leaves': 88, 'feature_fraction': 0.4537029051428528, 'min_child_samples': 25}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 21:17:04,391] Trial 33 finished with value: 0.8159406290223051 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5872409883213553, 'eta': 0.007405870921478667, 'num_iterations': 37569, 'lambda_l1': 2.8232682174433457, 'lambda_l2': 0.33103371803648324, 'num_leaves': 88, 'feature_fraction': 0.4537029051428528, 'min_child_samples': 25}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8159406290223051 0.5418976699468102 0.9026679599241426
filename ./models/lgbm_33 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5820485724914063, 'learning_rate': 0.007493396037960933, 'num_iterations': 36778, 'subsample_freq': 1, 'lambda_l1': 0.5019768434336962, 'lambda_l2': 0.3365610761306776, 'num_leaves': 117, 'feature_fraction': 0.42592546355478395, 'min_child_samples': 18}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 21:41:52,996] Trial 34 finished with value: 0.8154537413516078 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5820485724914063, 'eta': 0.007493396037960933, 'num_iterations': 36778, 'lambda_l1': 0.5019768434336962, 'lambda_l2': 0.3365610761306776, 'num_leaves': 117, 'feature_fraction': 0.42592546355478395, 'min_child_samples': 18}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8154537413516078 0.5412229927485859 0.9022232679558345
filename ./models/lgbm_34 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5786686492005797, 'learning_rate': 0.010206758549020305, 'num_iterations': 34755, 'subsample_freq': 1, 'lambda_l1': 2.506224854597924, 'lambda_l2': 0.07071316895661084, 'num_leaves': 39, 'feature_fraction': 0.4479920638729297, 'min_child_samples': 45}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 22:07:52,233] Trial 35 finished with value: 0.8158328279049585 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5786686492005797, 'eta': 0.010206758549020305, 'num_iterations': 34755, 'lambda_l1': 2.506224854597924, 'lambda_l2': 0.07071316895661084, 'num_leaves': 39, 'feature_fraction': 0.4479920638729297, 'min_child_samples': 45}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8158328279049585 0.5410907404474383 0.9026977594756362
filename ./models/lgbm_35 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5858001913044534, 'learning_rate': 0.014963486413133473, 'num_iterations': 37831, 'subsample_freq': 1, 'lambda_l1': 0.3263372308411499, 'lambda_l2': 0.0027411645206884493, 'num_leaves': 86, 'feature_fraction': 0.549286209233687, 'min_child_samples': 58}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 22:18:40,653] Trial 36 finished with value: 0.8148842012514935 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5858001913044534, 'eta': 0.014963486413133473, 'num_iterations': 37831, 'lambda_l1': 0.3263372308411499, 'lambda_l2': 0.0027411645206884493, 'num_leaves': 86, 'feature_fraction': 0.549286209233687, 'min_child_samples': 58}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8148842012514935 0.5423222215366673 0.9017611920561448
filename ./models/lgbm_36 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5737091401218981, 'learning_rate': 0.008405904035641693, 'num_iterations': 33334, 'subsample_freq': 1, 'lambda_l1': 0.023557698418466482, 'lambda_l2': 1.0423061646714944, 'num_leaves': 56, 'feature_fraction': 0.40266157500626626, 'min_child_samples': 29}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 22:39:27,796] Trial 37 finished with value: 0.8155200899701329 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5737091401218981, 'eta': 0.008405904035641693, 'num_iterations': 33334, 'lambda_l1': 0.023557698418466482, 'lambda_l2': 1.0423061646714944, 'num_leaves': 56, 'feature_fraction': 0.40266157500626626, 'min_child_samples': 29}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8155200899701329 0.5419979310024404 0.9023354333005699
filename ./models/lgbm_37 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5908701351213503, 'learning_rate': 0.0058825298705675, 'num_iterations': 32418, 'subsample_freq': 1, 'lambda_l1': 0.056537876157297516, 'lambda_l2': 0.23817340025196476, 'num_leaves': 111, 'feature_fraction': 0.5903891528884171, 'min_child_samples': 20}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 23:09:19,200] Trial 38 finished with value: 0.8154408944938037 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5908701351213503, 'eta': 0.0058825298705675, 'num_iterations': 32418, 'lambda_l1': 0.056537876157297516, 'lambda_l2': 0.23817340025196476, 'num_leaves': 111, 'feature_fraction': 0.5903891528884171, 'min_child_samples': 20}. Best is trial 21 with value: 0.8159660099828953.


Trial LGBM AUC 0.8154408944938037 0.5423375850112588 0.902360917164229
filename ./models/lgbm_38 best_model_name./models/best_lgbm ; best_trial_number 21
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5641972147140464, 'learning_rate': 0.007347787440390321, 'num_iterations': 37743, 'subsample_freq': 1, 'lambda_l1': 0.7809533651443774, 'lambda_l2': 0.04749261940282875, 'num_leaves': 21, 'feature_fraction': 0.4633328194247181, 'min_child_samples': 39}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-17 23:49:53,132] Trial 39 finished with value: 0.8161146771285536 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5641972147140464, 'eta': 0.007347787440390321, 'num_iterations': 37743, 'lambda_l1': 0.7809533651443774, 'lambda_l2': 0.04749261940282875, 'num_leaves': 21, 'feature_fraction': 0.4633328194247181, 'min_child_samples': 39}. Best is trial 39 with value: 0.8161146771285536.


Trial LGBM AUC 0.8161146771285536 0.5395740933702102 0.9028642837436492
filename ./models/lgbm_39 best_model_name./models/best_lgbm ; best_trial_number 39
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5638177768312883, 'learning_rate': 0.009208784700868814, 'num_iterations': 24312, 'subsample_freq': 1, 'lambda_l1': 0.8849807552531815, 'lambda_l2': 0.007449926605562585, 'num_leaves': 16, 'feature_fraction': 0.8715899551870319, 'min_child_samples': 50}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-18 00:19:57,036] Trial 40 finished with value: 0.8157762617451916 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5638177768312883, 'eta': 0.009208784700868814, 'num_iterations': 24312, 'lambda_l1': 0.8849807552531815, 'lambda_l2': 0.007449926605562585, 'num_leaves': 16, 'feature_fraction': 0.8715899551870319, 'min_child_samples': 50}. Best is trial 39 with value: 0.8161146771285536.


Trial LGBM AUC 0.8157762617451916 0.5406528613328756 0.9027202518824334
filename ./models/lgbm_40 best_model_name./models/best_lgbm ; best_trial_number 39
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.573099085855387, 'learning_rate': 0.007244632851899241, 'num_iterations': 37447, 'subsample_freq': 1, 'lambda_l1': 3.5389432163435335, 'lambda_l2': 0.05401143320044411, 'num_leaves': 34, 'feature_fraction': 0.4646475688812316, 'min_child_samples': 40}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-18 00:45:51,696] Trial 41 finished with value: 0.8159626259873438 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.573099085855387, 'eta': 0.007244632851899241, 'num_iterations': 37447, 'lambda_l1': 3.5389432163435335, 'lambda_l2': 0.05401143320044411, 'num_leaves': 34, 'feature_fraction': 0.4646475688812316, 'min_child_samples': 40}. Best is trial 39 with value: 0.8161146771285536.


Trial LGBM AUC 0.8159626259873438 0.5410004844425926 0.9025751429349581
filename ./models/lgbm_41 best_model_name./models/best_lgbm ; best_trial_number 39
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.572636020426708, 'learning_rate': 0.0074542698379182265, 'num_iterations': 37562, 'subsample_freq': 1, 'lambda_l1': 0.19494846317268774, 'lambda_l2': 0.06463389018961098, 'num_leaves': 29, 'feature_fraction': 0.4628614480374939, 'min_child_samples': 42}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.

[I 2021-09-18 01:21:52,439] Trial 42 finished with value: 0.8159337125393676 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.572636020426708, 'eta': 0.0074542698379182265, 'num_iterations': 37562, 'lambda_l1': 0.19494846317268774, 'lambda_l2': 0.06463389018961098, 'num_leaves': 29, 'feature_fraction': 0.4628614480374939, 'min_child_samples': 42}. Best is trial 39 with value: 0.8161146771285536.


Trial LGBM AUC 0.8159337125393676 0.5403548117836003 0.9027852386674697
filename ./models/lgbm_42 best_model_name./models/best_lgbm ; best_trial_number 39
STARTING Params: {'metric': 'AUC', 'verbosity': 0, 'device': 'gpu', 'boosting_type': 'gbdt', 'subsample': 0.5724045336241672, 'learning_rate': 0.00751375927856499, 'num_iterations': 37487, 'subsample_freq': 1, 'lambda_l1': 0.1203929410045197, 'lambda_l2': 0.03988094618438434, 'num_leaves': 34, 'feature_fraction': 0.46206876729543683, 'min_child_samples': 61}


/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:574: UserWarning:

Found `num_iterations` in params. Will use it instead of argument

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:578: UserWarning:

'early_stopping_rounds' argument is deprecated and will be removed in a future release of LightGBM. Pass 'early_stopping()' callback via 'callbacks' argument instead.



Trial 39 finished with value: 0.8161146771285536 and parameters: {'boosting_type': 'gbdt', 'subsample': 0.5641972147140464, 'eta': 0.007347787440390321, 'num_iterations': 37743, 'lambda_l1': 0.7809533651443774, 'lambda_l2': 0.04749261940282875, 'num_leaves': 21, 'feature_fraction': 0.4633328194247181, 'min_child_samples': 39}. Best is trial 39 with value: 0.8161146771285536.

Trial LGBM AUC 0.8161146771285536 0.5395740933702102 0.9028642837436492

Kaggle public score:  0.81790

In [ ]:
 # To get the dictionary of parameter name and parameter values:
 # Right now logging is not working as expected so this is pending
#logging.info(f'Return a dictionary of parameter name and parameter values:{lgbm_study.best_params}')

results = []
print(f'Return a dictionary of parameter name and parameter values:{lgbm_study.best_params}')
results.append({'msg':lgbm_study.best_params})
 
# To get the best observed value of the objective function:
#logging.info(f'Return the best observed value of the objective function: {lgbm_study.best_value}')
print(f'Return the best observed value of the objective function: {lgbm_study.best_value}')
results.append({'msg':lgbm_study.best_value})
 
# To get the best trial:
print(f'Return the best trial: {lgbm_study.best_trial}')
results.append({'msg': lgbm_study.best_trial})

pd.DataFrame(results).to_csv('results.csv', index=False)
 
# To get all trials:
#print("Return all the trials:", study.trials)

NameError: ignored

In [ ]:
model_file = open('./models/best_lgbm', 'rb')
colab_lgbm_model_eval = pickle.load(model_file)
model_file.close()

In [ ]:
colab_lgbm_model_eval.fit(X,y)
y_predproba = colab_lgbm_model_eval.predict_proba(X)

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_label.py:235: DataConversionWarning:

A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().

/usr/local/lib/python3.7/dist-packages/sklearn/preprocessing/_label.py:268: DataConversionWarning:

A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples, ), for example using ravel().

/usr/local/lib/python3.7/dist-packages/lightgbm/engine.py:177: UserWarning:

Found `num_iterations` in params. Will use it instead of argument



[LightGBM] [Warning] feature_fraction is set=0.4633328194247181, colsample_bytree=1.0 will be ignored. Current value: feature_fraction=0.4633328194247181
[LightGBM] [Warning] lambda_l2 is set=0.04749261940282875, reg_lambda=0.0 will be ignored. Current value: lambda_l2=0.04749261940282875
[LightGBM] [Warning] lambda_l1 is set=0.7809533651443774, reg_alpha=0.0 will be ignored. Current value: lambda_l1=0.7809533651443774
[LightGBM] [Warning] feature_fraction is set=0.4633328194247181, colsample_bytree=1.0 will be ignored. Current value: feature_fraction=0.4633328194247181
[LightGBM] [Warning] lambda_l2 is set=0.04749261940282875, reg_lambda=0.0 will be ignored. Current value: lambda_l2=0.04749261940282875
[LightGBM] [Warning] lambda_l1 is set=0.7809533651443774, reg_alpha=0.0 will be ignored. Current value: lambda_l1=0.7809533651443774


In [ ]:
roc_auc_score(y, y_predproba[:, 1])

0.9017065674757444

In [ ]:
y_predproba = pd.DataFrame(colab_lgbm_model_eval.predict_proba(test_X))
y_predproba['id'] = y_predproba.index + 957919
y_predproba.to_csv('./artifacts/lightgbm_predproba.csv', index=False)
results_df = y_predproba.rename(columns={1:'claim'}).copy()
results_df = results_df[['id','claim']]
results_df.to_csv('./artifacts/lightgbm_submission.csv', index=False)